src/btree/bt_curnext.c

```diff
@@ -677,6 +679,23 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     for (newpage = false;; newpage = true, restart = false) {
         page = cbt->ref == NULL ? NULL : cbt->ref->page;
 
+        /*
+         * Determine if all records on the page have been deleted and all the tombstones are visible
+         * to our transaction. If so, we can avoid reading the records on the page and move to the
+         * next page. We base this decision on the aggregate stop point added to the page during the
+         * last reconciliation. We can skip this test if the page has been modified since it was
+         * reconciled or the underlying cursor is configured to ignore tombstones.
+         */
+        if (session->txn->isolation == WT_ISO_SNAPSHOT &&
```

Comment:
I think we need to exclude this optimization for history store? The history store dump with globally visible tombstone will be broken by this change I think.